src/header/index.tsx

```diff
@@ -0,0 +1,63 @@
+import * as css from '../theme/default/header.m.css';
+import theme from '../middleware/theme';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { ThemedProperties } from '@dojo/framework/core/mixins/Themed';
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+export interface HeaderProperties extends ThemedProperties {
+	/** Determines if this header is fixed */
+	sticky?: boolean;
+}
+
+export type HeaderChildren =
+	| {
+			/** Renderer for leading elements like icons */
+			leading?(): RenderResult;
+			/** Renderer for the header title */
+			title(): RenderResult;
+			/** Renderer for header actions like links */
+			actions?(): RenderResult;
+			/** Renderer for trailing elements like search inputs */
+			trailing?(): RenderResult;
+	  }
+	| undefined;
+
+const factory = create({ theme })
+	.properties<HeaderProperties>()
+	.children<HeaderChildren>();
+
+export const Header = factory(function Header({ children, properties, middleware: { theme } }) {
+	const classes = theme.classes(css);
+	const { sticky } = properties();
+	const { actions = undefined, leading = undefined, title = undefined, trailing = undefined } =
+		children()[0] || {};
+
+	const actionElements = actions && actions();
+
+	return (
+		<header key="header" classes={classes.spacer}>
```

Comment:
does `classes.spacer` always need to be applied, or just when the widget is `sticky` ?